tensorflow_datasets/scripts/random_image_generator.py

```diff
@@ -0,0 +1,102 @@
+import os
+from zipfile import ZipFile
+from PIL import Image
+import numpy as np
+import tarfile
+import zipfile
+import shutil
+import gzip
+
+
+# images_path variable to store the path of images for later compression
+images_path = []
+
+
+def image_process(root):
+    """ Generate random images and remove noise of the
+        images to compress better.
+
+          Args:
+           root: path of the images to get processed
+
+    """
+    images_path.append(root)
+    image = np.array(Image.open(root))
+    image = np.ones_like(image) * np.random.randint(np.amax(image))
+    image = Image.fromarray(image)
+    image = image.convert('RGB')
+    image.save(root)
+
+
+def rewrite_zip(filepath, filename):
+    """ Rewrite the older .zip files into new better compressed one
+
+        Args:
+            filepath: path of the older zip file that need to be compressed
+            filename: name of the older zip file that need to be compressed
+
+    """
+    # creating a temporary file to store images 
+    temp = tempfile.TemporaryDirectory(dir=filepath)
```

Comment:
`with tempfile.TemporaryDirectory(dir=filepath)` to automatically cleanup the temporary file afterward

Reply:
> Thank you for the PR. Could you also send another PR of the script applied on our fake_data dir to see what the result is ?

@Conchylicultor  : Sorry but i want to know something more about this. Could you share me the link of fake_data dir.  And what i need to do in that directory. 
Thank you

Reply:
How did you tested your script ? The goal of this issue is to compress the size of our fake data dir: https://github.com/tensorflow/datasets/tree/master/tensorflow_datasets/testing/test_data/fake_examples

You should test the script on this folder to make sure this works. Also could you report the directory size before/after image compression ?

Reply:
I have tested the script with my own data. But i will report you the results within some minutes.

Reply:
Awesome thank you.

Reply:
So i need to compress the whole directory or i just check my script on one file of that directory.

Reply:
Try to use the script on the full directory to make sure this works as expected. Thank you.